doc/source/ray-air/examples/tensorflow_image_example.md

```diff
@@ -0,0 +1,265 @@
+---
+jupytext:
+    text_representation:
+        extension: .md
+        format_name: myst
+kernelspec:
+    display_name: Python 3
+    language: python
+    name: python3
+---
+
+
+# Training a TensorFlow Classifier
+
+This tutorial demonstrates how to train an image classifier using the [Ray AI Runtime](air) (AIR).
+
+You should be familiar with TensorFlow before starting this tutorial. If you need a refresher, read TensorFlow's [Convolutional Neural Network](https://www.tensorflow.org/tutorials/images/cnn) tutorial.
+
+## Before you begin
+
+* Install the [Ray AI Runtime](air). You'll need Ray 1.13 or later to run this example.
+
+```{code-cell} python3
+!pip install 'ray[air]'
+```
+
+* Install `tensorflow` and `tensorflow-datasets`
+
+```{code-cell} python3
+!pip install tensorflow tensorflow-datasets
+```
+
+## Load and normalize CIFAR-10
+
+We'll train our classifier on a popular image dataset called [CIFAR-10](https://www.cs.toronto.edu/~kriz/cifar.html).
+
+First, let's load CIFAR-10 into a Ray Dataset.
+
+```{code-cell} ipython3
+import ray
+from ray.data.datasource import SimpleTensorFlowDatasource
+import tensorflow as tf
+
+from tensorflow.keras import layers, models
+import tensorflow_datasets as tfds
+
+
+def train_dataset_factory():
+    return tfds.load("cifar10", split=["train"], as_supervised=True)[0]
+
+def test_dataset_factory():
+    return tfds.load("cifar10", split=["test"], as_supervised=True)[0]
+
+train_dataset = ray.data.read_datasource(
+    SimpleTensorFlowDatasource(), dataset_factory=train_dataset_factory
+)
+test_dataset = ray.data.read_datasource(
+    SimpleTensorFlowDatasource(), dataset_factory=test_dataset_factory
+)
+
+train_dataset
+```
+
+Note that {py:class}`SimpleTensorFlowDatasource <ray.data.datasource.SimpleTensorFlowDatasource>` loads all data into memory, so you shouldn't use it with larger datasets.
+
+Our model will expect float arrays, so let's normalize pixel values to be between 0 and 1.
+
+```{code-cell} ipython3
+def normalize_images(batch):
+    return [(tf.cast(image, tf.float32) / 255.0, label) for image, label in batch]
+
+train_dataset = train_dataset.map_batches(normalize_images)
+test_dataset = test_dataset.map_batches(normalize_images)
+```
+
+Next, let's represent our data using pandas dataframes instead of tuples. This lets us call methods like {py:meth}`Dataset.to_tf <ray.data.Dataset.to_tf>` later in the tutorial.
+
+```{code-cell} python3
+import pandas as pd
+from ray.data.extensions import TensorArray
+
+
+def convert_batch_to_pandas(batch):
+    images = TensorArray([image.numpy() for image, _ in batch])
+    labels = [label.numpy() for _, label in batch]
```

Comment:
Labels are just normal ints. I felt that it was simpler to keep the labels as ints.

Reply:
I'm surprised that this doesn't work as expected, then, since I thought that `EagerTensor.numpy()` would return a scalar when representing a scalar tensor. 🤔